xdsl/parser.py

```diff
@@ -1790,6 +1790,22 @@ def _parse_op_args_list(self) -> list[Span]:
         # TODO: check if type is correct here!
         return args
 
+    def parse_region_list(self) -> list[Region]:
+        """
+        Parses a sequence of regions for as long as there is a `{` in the input.
+        """
+        regions: list[Region] = []
+        while not self.tokenizer.is_eof() and self.tokenizer.starts_with("{"):
+            regions.append(self.parse_region())
+            if self.tokenizer.starts_with(','):
+                self.parse_characters(',',
+                                      msg='This error should never be printed')
```

Comment:
Can you give more constructive feedbakc? Something like: "Please submit an issue if you run into this error" or something? Maybe even with a link to our issue tracker?